src/ocean/text/convert/Integer.d

```diff
@@ -30,6 +30,138 @@ import ocean.transition;
 
 import ocean.core.Traits;
 
+import ocean.core.array.Search : find;
+
+import ocean.util.ReusableException;
+
+import ocean.math.Math;
+
+
+/*******************************************************************************
+
+    Parses a floating point number represented as a string directly to an
+    integer value.
+
+    To represent the fractional part we multiply the value by the requested
+    amount of decimal points and add it up. For example:
+
+    "1.123" -> 1123  (decimal_points = 3)
+    "0.01"  ->   10  (decimal_points = 3)
+
+    Any characters longer than the requested amount of decimal points will be
+    cut off:
+
+    "1.2345"  ->  123 (decimal_points = 2)
+    "10.2030" -> 1020 (decimal_points = 2)
+
+    Params:
+        decimal_points = amount of decimal points to consider
+        T              = type of the integer
+        value          = floating point number string to parse
+        exception      = optional, a reusable exception instance to throw
+
+    Returns:
+        the resulting integer number
+
+*******************************************************************************/
+
+public T floatStringToInt ( T = ulong ) ( size_t decimal_points, cstring value, ReusableException exception = null )
```

Comment:
I notice the API of this function is at variance with all the other functions in this module -- it returns the converted value and throws on error. All the other functions return a bool indicating parsing success and have an out value to return the converted integer. It would seem cleaner to follow the same pattern with this new function.